swarm/pss/client/client.go

```diff
@@ -9,58 +9,77 @@ import (
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/event"
 	"github.com/ethereum/go-ethereum/log"
-	"github.com/ethereum/go-ethereum/node"
 	"github.com/ethereum/go-ethereum/p2p"
 	"github.com/ethereum/go-ethereum/p2p/discover"
 	"github.com/ethereum/go-ethereum/p2p/protocols"
 	"github.com/ethereum/go-ethereum/pot"
+	"github.com/ethereum/go-ethereum/rlp"
 	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/ethereum/go-ethereum/swarm/pss"
 )
 
 const (
 	inboxCapacity  = 3000
 	outboxCapacity = 100
+	defaultWSHost  = 8546
 	addrLen        = common.HashLength
 )
 
+// RemoteHost: hostname of node running websockets proxy to pss (default localhost)
+// RemotePort: port of node running websockets proxy to pss (0 = go-ethereum node default)
+// Secure: whether or not to use secure connection
+// SelfHost: local if host to connect from
+type PssClientConfig struct {
+	SelfHost   string
+	RemoteHost string
+	RemotePort int
+	Secure     bool
+}
+
+func NewPssClientConfig() *PssClientConfig {
+	return &PssClientConfig{
+		SelfHost:   "localhost",
+		RemoteHost: "localhost",
+		RemotePort: 8546,
+	}
+}
+
 type PssClient struct {
 	localuri     string
 	remoteuri    string
 	ctx          context.Context
 	cancel       func()
 	subscription *rpc.ClientSubscription
 	topicsC      chan []byte
-	msgC         chan PssAPIMsg
+	msgC         chan pss.PssAPIMsg
 	quitC        chan struct{}
 	quitting     uint32
 	ws           *rpc.Client
 	lock         sync.Mutex
-	peerPool     map[PssTopic]map[pot.Address]*pssRPCRW
-	protos       []*p2p.Protocol
+	peerPool     map[pss.PssTopic]map[pot.Address]*pssRPCRW
+	protos       map[pss.PssTopic]*p2p.Protocol
```

Comment:
as suggested on #100 let;s get rid of the Pss prefixes, will make the code more readable
eg. `pss.Topic, pss.API`